app/src/main.ts

```diff
@@ -159,6 +167,18 @@ if (shouldQuit) {
         });
       });
     }
+    if (!appArgs.disableOldNativefierWarning) {
+      (async () => {
+        if ((await latestVersion('nativefier')) != appArgs.nativefierVersion) {
+          dialog.showMessageBox(null, {
```

Comment:
I'm not a fan of showing a messagebox, I don't like the fact that it's blocking. I would have preferred something visible but not blocking. Here's what I suggested in https://github.com/jiahaog/nativefier/issues/556 :

> *"Maybe a top, bright, red bar, like Firefox does when a deceptive site is detected."*

Reply:
How would I make a bar at the top?

Reply:
> How would I make a bar at the top?

@mattruzzi using [Electron HTML & CSS](https://www.electronjs.org/docs/tutorial/quick-start#create-a-basic-application) and [Electron APIs](https://www.electronjs.org/docs/api). Right now, Nativefier is [just a BrowserWindow](https://github.com/jiahaog/nativefier/blob/3e0011a29cc28c2201358fb6e8afa685eab1e9b5/app/src/components/mainWindow.ts#L107) in which [we `loadURL` your webapp URL](https://github.com/jiahaog/nativefier/blob/3e0011a29cc28c2201358fb6e8afa685eab1e9b5/app/src/components/mainWindow.ts#L247). To do what I'm suggesting, I guess that:

A. Either you should replace the trivial thing we're doing with a container of two children stacked vertically: 1. a potential alert bar (hidden under normal conditions, or of fixed vertical size if alerting), and 2. the webapp container taking the rest of the space and showing the app.

B. Or maybe A is wrong, and maybe in Electron the right thing to do is to keep loading the user's URL in the main BrowserWindow as we're doing, but *inject* our alert header ourselves directly in the webapp HTML.

I don't have more details to bring. I were to build this, I'd have to go spelunking for docs and examples too, maybe asking around Electron forums/reddits, and determine what works 🙂.

Also, feel free to debate my point that the bar I'm suggesting is better than an alert.

Reply:
>     * Should be made in a way that cannot be mistaken with the bundled website. 
A bar at the top of the main window could be confused with something from the bundled site since the space at the top of the main window that the bar takes up would normally be the site. I don't think that a bundled website would be able to show that kind of alert box.